app/assets/javascripts/discourse/app/lib/bulk-select-helper.js

```diff
@@ -0,0 +1,66 @@
+import { NotificationLevels } from "discourse/lib/notification-levels";
+import Topic from "discourse/models/topic";
+import { inject as service } from "@ember/service";
+import { getOwner, setOwner } from "@ember/application";
+import { tracked } from "@glimmer/tracking";
+import { TrackedArray } from "@ember-compat/tracked-built-ins";
+
+export default class BulkSelectHelper {
+  @service router;
+  @service modal;
+  @service pmTopicTrackingState;
+  @service topicTrackingState;
+
+  @tracked bulkSelectEnabled = false;
+  @tracked autoAddTopicsToBulkSelect = false;
+
+  selected = new TrackedArray();
```

Comment:
Do I understand correctly that because it's a TrackedArray, we don't care about `@tracked` right?

Reply:
`@tracked` means "if `this.selected` is reassigned, notify any dependents". In this case, we know for sure that `selected` will never be reassigned, so there's no need for the `@tracked` overhead.

Any mutations of the TrackedArray instance will notify dependents :+1:

Reply:
Yes that's how I understand/use it. :+1: